docs/ECMAScriptModules.md

```diff
@@ -38,4 +38,57 @@ import.meta.jest.useFakeTimers();
 // jest === import.meta.jest => true
 ```
 
-Please note that we currently don't support `jest.mock` in a clean way in ESM, but that is something we intend to add proper support for in the future. Follow [this issue](https://github.com/facebook/jest/issues/10025) for updates.
+Additionally, since ESM evaluates static `import` statements before looking at the code, hoisting on `jest.mock` calls that happens in CJS modules won't work in ESM. To `mock` modules in ESM, you need to use `require` or dynamic `import()` after `jest.mock` calls to load the mocked modules, same applies to modules which have to load the mocked modules.
```

Comment:
Last idea from my side:

```suggestion

## Module mocking in ESM

Since ESM evaluates static `import` statements before looking at the code, hoisting on `jest.mock` calls that happens in CJS modules won't work in ESM. To `mock` modules in ESM, you need to use `require` or dynamic `import()` after `jest.mock` calls to load the mocked modules, same applies to modules which have to load the mocked modules.
```

Reply:
How about we have this title added on v27+ only? We can only `mock` CJS modules before v27 even with ESM support enabled right? Plus I've noticed that the `ECMAScript Modules` page doesn't show up on v25's sidebar. Is that list a generated one or we have to update the `sidebars.json` manually?

Reply:
Right, it makes sense only with v27+. I will take a look at the sidebars later. Can it be there was no ESM support in v25? No sure.

Reply:
Looked around quickly. Seems like first steps with ESM support were released in v25. So you are right here too, that link is missing in the sidebar and must be added. Thanks!